tee-worker/core-primitives/types/src/lib.rs

```diff
@@ -45,22 +45,42 @@ pub use itp_sgx_runtime_primitives::types::*;
 pub type IpfsHash = [u8; 46];
 pub type MrEnclave = [u8; 32];
 
+pub type CallIndex = [u8; 2];
+
 // pallet teerex
-pub type ConfirmCallFn = ([u8; 2], ShardIdentifier, H256, Vec<u8>);
-pub type ShieldFundsFn = ([u8; 2], Vec<u8>, Balance, ShardIdentifier);
-pub type CallWorkerFn = ([u8; 2], Request);
+pub type ConfirmCallFn = (CallIndex, ShardIdentifier, H256, Vec<u8>);
+pub type ShieldFundsFn = (CallIndex, Vec<u8>, Balance, ShardIdentifier);
+pub type CallWorkerFn = (CallIndex, Request);
 
 // pallet IMP
-pub type SetUserShieldingKeyFn = ([u8; 2], ShardIdentifier, Vec<u8>);
-pub type CreateIdentityFn = ([u8; 2], ShardIdentifier, AccountId, Vec<u8>, Option<Vec<u8>>);
-pub type RemoveIdentityFn = ([u8; 2], ShardIdentifier, Vec<u8>);
-pub type VerifyIdentityFn = ([u8; 2], ShardIdentifier, Vec<u8>, Vec<u8>);
+pub type SetUserShieldingKeyParameters = (ShardIdentifier, Vec<u8>);
+pub type SetUserShieldingKeyFn = (CallIndex, SetUserShieldingKeyParameters);
+
+pub type CreateIdentityParameters = (ShardIdentifier, AccountId, Vec<u8>, Option<Vec<u8>>);
+pub type CreateIdentityFn = (CallIndex, CreateIdentityParameters);
+
+pub type RemoveIdentityParameters = (ShardIdentifier, Vec<u8>);
+pub type RemoveIdentityFn = (CallIndex, RemoveIdentityParameters);
+
+pub type VerifyIdentityParameters = (ShardIdentifier, Vec<u8>, Vec<u8>);
+pub type VerifyIdentityFn = (CallIndex, VerifyIdentityParameters);
 
 // pallet VCMP
-pub type RequestVCFn = ([u8; 2], ShardIdentifier, Assertion);
+pub type RequestVCFn = (CallIndex, ShardIdentifier, Assertion);
+
+// pallet Utility
+pub type BatchAllFn = (CallIndex, Vec<SupportedCall>);
```

Comment:
So if I understand it correctly it works if the batch **only** contains the supported calls - maybe we can extend it later to allow more than this (but only extract and handle supported calls)

Reply:
Yes. Currently we only allow the SupportedCall.